src/test/ui/async-await/issue-64130-non-send-future-diags.stderr

```diff
@@ -0,0 +1,23 @@
+error[E0277]: `std::sync::MutexGuard<'_, u32>` cannot be sent between threads safely
+  --> $DIR/issue-64130-non-send-future-diags.rs:23:5
+   |
+LL | fn is_send<T: Send>(t: T) {
+   |    -------    ---- required by this bound in `is_send`
+...
+LL |     is_send(foo());
+   |     ^^^^^^^ `std::sync::MutexGuard<'_, u32>` cannot be sent between threads safely
```

Comment:
Hmm, I think that we may want to intercept this error earlier -- in particular I think we probably want to change *this* text too, so that it says "future cannot be sent between threads safely" or something like that. I'll have to check what the exact text that @yoshuawuyts and I came up with was. We could perhaps leave this for a future PR, though.

Reply:
> We could perhaps leave this for a future PR, though.

Let's leave this as a follow-up.